crates/noirc_evaluator/src/brillig/brillig_gen.rs

```diff
@@ -187,20 +202,75 @@ impl BrilligGen {
         &mut self,
         binary: &Binary,
         dfg: &DataFlowGraph,
+        // TODO: For array addition, perhaps this will need
+        // TODO to be multiple registers
         result_register: RegisterIndex,
     ) {
-        let binary_type =
-            type_of_binary_operation(dfg[binary.lhs].get_type(), dfg[binary.rhs].get_type());
+        let lhs_type = dfg[binary.lhs].get_type();
+        let rhs_type = dfg[binary.rhs].get_type();
 
         let left = self.convert_ssa_value(binary.lhs, dfg);
         let right = self.convert_ssa_value(binary.rhs, dfg);
 
+        let binary_type = type_of_binary_operation(lhs_type, rhs_type);
+        // For operations involving arrays, we handle as a special case
+        if let Type::Array(element_type, num_elements) = binary_type {
+            return self.convert_ssa_binary_array(
+                &element_type,
+                left,
+                right,
+                num_elements,
+                binary.operator,
+                dfg,
+                result_register,
+            );
+        }
+
         let brillig_binary_op =
             convert_ssa_binary_op_to_brillig_binary_op(binary.operator, binary_type);
 
         self.context.binary_instruction(left, right, result_register, brillig_binary_op);
     }
 
+    /// Handles binary operations that involve arrays as the operands.
+    fn convert_ssa_binary_array(
+        &mut self,
+        element_type: &[Type],
+        lhs_array_ptr: RegisterIndex,
+        rhs_array_ptr: RegisterIndex,
+        num_elements: usize,
+        operator: BinaryOp,
+        dfg: &DataFlowGraph,
+        // TODO: For array addition, perhaps this will need
+        // TODO to be multiple registers
+        result_register: RegisterIndex,
+    ) {
+        // For now, we only support arrays of single element types
+        assert_eq!(
+            element_type.len(),
+            1,
+            "array has a composite type as its element, which we do not support yet"
+        );
+        let atomic_type = element_type[0].clone();
+
+        match operator {
+            BinaryOp::Eq => {
+                // TODO:
+                // Add a for loop which does roughly
```

Comment:
Currently this code returns 1 for array equality;  just need to implement the for-loop noted below